In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('/content/House Price India.csv')

In [3]:
df.head()

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
0,6762810145,42491,5,2.50,3650,9050,2.0,0,4,5,...,1921,0,122003,52.8645,-114.557,2880,5400,2,58,2380000
1,6762810635,42491,4,2.50,2920,4000,1.5,0,0,5,...,1909,0,122004,52.8878,-114.470,2470,4000,2,51,1400000
2,6762810998,42491,5,2.75,2910,9480,1.5,0,0,3,...,1939,0,122004,52.8852,-114.468,2940,6600,1,53,1200000
3,6762812605,42491,4,2.50,3310,42998,2.0,0,0,3,...,2001,0,122005,52.9532,-114.321,3350,42847,3,76,838000
4,6762812919,42491,3,2.00,2710,4500,1.5,0,0,4,...,1929,0,122006,52.9047,-114.485,2060,4500,1,51,805000


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 23 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     14620 non-null  int64  
 1   Date                                   14620 non-null  int64  
 2   number of bedrooms                     14620 non-null  int64  
 3   number of bathrooms                    14620 non-null  float64
 4   living area                            14620 non-null  int64  
 5   lot area                               14620 non-null  int64  
 6   number of floors                       14620 non-null  float64
 7   waterfront present                     14620 non-null  int64  
 8   number of views                        14620 non-null  int64  
 9   condition of the house                 14620 non-null  int64  
 10  grade of the house                     14620 non-null  int64  
 11  Ar

In [5]:
df.describe()

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
count,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,...,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04
mean,6.762821e+09,42604.538646,3.379343,2.129583,2098.262996,1.509328e+04,1.502360,0.007661,0.233105,3.430506,...,1970.926402,90.924008,122033.062244,52.792848,-114.404007,1996.702257,12753.500068,2.012244,64.950958,5.389322e+05
std,6.237575e+03,67.347991,0.938719,0.769934,928.275721,3.791962e+04,0.540239,0.087193,0.766259,0.664151,...,29.493625,416.216661,19.082418,0.137522,0.141326,691.093366,26058.414467,0.817284,8.936008,3.675324e+05
min,6.762810e+09,42491.000000,1.000000,0.500000,370.000000,5.200000e+02,1.000000,0.000000,0.000000,1.000000,...,1900.000000,0.000000,122003.000000,52.385900,-114.709000,460.000000,651.000000,1.000000,50.000000,7.800000e+04
25%,6.762815e+09,42546.000000,3.000000,1.750000,1440.000000,5.010750e+03,1.000000,0.000000,0.000000,3.000000,...,1951.000000,0.000000,122017.000000,52.707600,-114.519000,1490.000000,5097.750000,1.000000,57.000000,3.200000e+05
50%,6.762821e+09,42600.000000,3.000000,2.250000,1930.000000,7.620000e+03,1.500000,0.000000,0.000000,3.000000,...,1975.000000,0.000000,122032.000000,52.806400,-114.421000,1850.000000,7620.000000,2.000000,65.000000,4.500000e+05
75%,6.762826e+09,42662.000000,4.000000,2.500000,2570.000000,1.080000e+04,2.000000,0.000000,0.000000,4.000000,...,1997.000000,0.000000,122048.000000,52.908900,-114.315000,2380.000000,10125.000000,3.000000,73.000000,6.450000e+05
max,6.762832e+09,42734.000000,33.000000,8.000000,13540.000000,1.074218e+06,3.500000,1.000000,4.000000,5.000000,...,2015.000000,2015.000000,122072.000000,53.007600,-113.505000,6110.000000,560617.000000,3.000000,80.000000,7.700000e+06


**Q1. What is the average, median, and standard deviation of house prices?**

In [6]:
# Average Price

df['Price'].mean()

np.float64(538932.2183310534)

In [7]:
# Median Price

df['Price'].median()

450000.0

In [8]:
# Standard Deviation

df['Price'].std()

367532.3808039669

**Q2. Which number of bedrooms is most common?**

In [9]:
print ('No. of bedrooms most common')
df['number of bedrooms'].mode()[0]

No. of bedrooms most common


np.int64(3)

**Q3. Check if the price data is skewed.**

In [10]:
skewness= df['Price'].skew()
print ('Skewness:',skewness)

Skewness: 4.269297720707116


In [11]:
if skewness >0 :
  print ('Price data is positively skewed')
elif skewness <0:
  print ('Price data is negatively skewed')
else :
  print ('Price data is normally distributed')


Price data is positively skewed


**Q4. What is the average price for each number of bedrooms?**

In [12]:
df.groupby('number of bedrooms')['Price'].mean().round(2)

,Price
number of bedrooms,
1,308963.75
2,398547.57
3,463277.56
4,636198.81
5,775255.03
6,837581.48
7,1016543.90
8,1208454.55
9,776666.33


**Q5.  What is the relationship between living area and price? (Use : Area without Basement).**

In [38]:
df[['Area of the house(excluding basement)','Price']].corr()

,Area of the house(excluding basement),Price
Area of the house(excluding basement),1.00000,0.61522
Price,0.61522,1.00000


**Q6. Identify any anomalies where houses have high prices but low areas.**

In [39]:
df[['Area of the house(excluding basement)','Price']].sort_values(['Price','Area of the house(excluding basement)'], ascending=[False,True]).head(10)

,Area of the house(excluding basement),Price
10503,8570,7700000
2424,7680,7060000
9171,8860,6890000
6244,6200,5570000
10915,5990,5110000
2794,4820,4670000
2907,6430,4490000
10566,5760,4000000
12442,5770,3850000
3946,4320,3800000


These houses have relatively high prices despite having comparatively smaller areas, so they can be considered potential price area anomalies.


**Q7. Compare average price based on number of floors and houses with or without waterfront.**

In [34]:
print('Average price by number of floors and waterfront presence:')
display(df.groupby(['number of floors' , 'waterfront present'])['Price'].mean().unstack().round(2))

Average price by number of floors and waterfront presence:


waterfront present,0,1
number of floors,,
1.0,432372.39,1211080.95
1.5,547616.22,1232506.57
2.0,635774.21,2075931.37
2.5,1087653.76,3280000.00
3.0,555809.50,1655000.00
3.5,1205875.00,NaN


**Q8. Identify the minimum and maximum house price. What does this indicate?**

In [17]:
df['Price'].min()

78000

In [18]:
df['Price'].max()

7700000

The minimum house price of `78000` and the maximum house price of `7700000` indicate a very wide range in property values within this dataset. This suggests a diverse housing market, likely including properties from various segments, such as small, affordable homes to very large, luxurious estates. This broad price range is important to consider when performing further analysis, as it implies significant variation in factors influencing house prices.

**Q9. Which location (zipcode/area) has the highest average price?**

In [19]:
df.groupby('Postal Code')['Price'].mean().round(2).sort_values(ascending=False).head(1)

,Price
Postal Code,
122071,2348310.81


**Q10. Write at least 5 insights from your analysis.**

### Key Insights:

1.  **Skewed Prices**: House prices are positively skewed, meaning most houses are lower-priced with a few very expensive ones.
2.  **Area-Price Correlation**: Larger living areas (excluding basement) generally lead to higher house prices (correlation ~0.615).
3.  **Bedroom Impact**: More bedrooms usually correlate with higher average house prices, with 3-bedroom homes being most common.
4.  **Geographic Variation**: Prices vary significantly by location, with Postal Code 122071 having the highest average price.
5.  **Anomalies**: Some houses have unusually high prices for their small size, indicating other non-area factors like location or unique features heavily influence their value.